test/transaction.test.js

```diff
@@ -173,31 +173,37 @@ describe('transactions', function() {
   });
 
   describe('timeout', function() {
+    const TIMEOUT = 50;
     before(function() {
       // Reset the collection
       db.connector.data = {};
     });
 
     var post = {title: 't3', content: 'c3'};
-    before(createPostInTx(post, 50));
+    beforeEach(createPostInTx(post, TIMEOUT));
 
     it('should report timeout', function(done) {
-      setTimeout(function() {
```

Comment:
This test did not make any sense to me. The original code:

```js
    it('should report timeout', function(done) {
      setTimeout(function() {
        Post.find({where: {title: 't3'}}, {transaction: currentTx},
          function(err, posts) {
            if (err) return done(err);
            expect(posts.length).to.be.eql(1);
            done();
          });
      }, 300);
      done();
    });
```

Notice the `done()` callback is called immediately after `setTimeout`, before `setTimeout` fires. As a result, the test was marked as passed by mocha and because the remaining tests finished before the timeout fired, the process exited. 

Mocha no longer exits the process when tests finish, in which case the `setTimeout` callback has time to be fired and fails with an error - this error is reported outside of any test case though. What a mess!

I rewrote the test as follows, I hope I captured author's intent correctly (ping @raymondfeng):

```js
    it('should report timeout', function(done) {
      // wait until the "create post" transaction times out
      setTimeout(runTheTest, TIMEOUT * 3);

      function runTheTest() {
        Post.find({where: {title: 't3'}}, {transaction: currentTx},
          function(err, posts) {
            expect(err).to.match(/transaction.*not active/);
            done();
          });
      }
    });
```

Reply:
Good catch.